### PR TITLE
[CI] green llama tests

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Before start the work to refactor models, let's make the tests on our base model green 🤗 All tests on llama are green after this PR, except for the flex attention tests (which is a WIP feature)

Fixes:
- `batch_size` -> `max_batch_size` in `StaticCache` (#37007)
- `generate` + FA2 test -> don't pass attention masks with right-padding (FA2 doesn't support all attention mask patterns, and with generate the mask would have holes like `1 1 0 0 0 0 1` at generation time)
- models with compilation integration tests -> clear CUDA cache (see comment in `LlamaIntegrationTest.tearDown`)

### CODE REVIEW

File: tests/generation/test_utils.py [hunk 1]
```diff
@@ -2285,6 +2285,12 @@ def _test_attention_implementation(self, attn_implementation):
                     inputs_dict[input_name] = input_data
             main_input = inputs_dict[model_class.main_input_name]
 
+            # FA2 doesn't accept masking in the middle of the sequence for now
+            if attn_implementation == "flash_attention_2":
+                for input_name in ("attention_mask", "decoder_attention_mask", "encoder_attention_mask"):
+                    if input_name in inputs_dict:
+                        inputs_dict[input_name] = torch.ones_like(inputs_dict[input_name])
```
Comment (ydshieh) [inline tests/generation/test_utils.py:2292]: I am not able to see easily the relation of this change and the comment

> generate + FA2 test -> don't pass attention masks with right-padding (they are only equivalent with left-padding)

if you can explain it a bit more for me 🙏 .

File: tests/models/deepseek_v3/test_modeling_deepseek_v3.py [hunk 2]
```diff
@@ -605,6 +606,10 @@ def setUpClass(cls):
             # 8 is for A100 / A10 and 7 for T4
             cls.cuda_compute_capability_major_version = torch.cuda.get_device_capability()[0]
 
+    def tearDown(self):
+        # See LlamaIntegrationTest.tearDown(). Can be removed once LlamaIntegrationTest.tearDown() is removed.
+        cleanup(torch_device, gc_collect=False)
```
Comment (ydshieh) [inline tests/models/deepseek_v3/test_modeling_deepseek_v3.py:611]: For compilation, it seems to me better to also include

> torch._dynamo.reset()

see

https://github.com/pytorch/FBGEMM/pull/1992/files

If you agree, you can simply add a new key argument to `def cleanup`

**Review by ydshieh — APPROVED** [review]
LGTM, thanks. Just 2 nits but up to you

File: tests/generation/test_utils.py (hunk 1 above)
Comment (gante) [reply tests/generation/test_utils.py:2292]: FA2 doesn't support all types of attention masks, and the support depends on the version of FA2. See `flash_attn_supports_top_left_mask` 👀 

The `attention_mask` in the tests is often right-padded, e.g. `input_mask = torch.tril(torch.ones_like(input_ids).to(torch_device))`

File: tests/models/deepseek_v3/test_modeling_deepseek_v3.py (hunk 2 above)
Comment (gante) [reply tests/models/deepseek_v3/test_modeling_deepseek_v3.py:611]: ah yes, good idea -- going to add to this PR!

File: tests/generation/test_utils.py (hunk 1 above)
Comment (gante) [reply tests/generation/test_utils.py]: (also added comment to test)
